azalea-auth/src/game_profile.rs

```diff
@@ -24,3 +25,146 @@ pub struct ProfilePropertyValue {
     pub value: String,
     pub signature: Option<String>,
 }
+
+impl Serialize for GameProfile {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut serializer = serializer.serialize_struct("GameProfile", 3)?;
+        serializer.serialize_field("id", &self.uuid)?;
+        serializer.serialize_field("name", &self.name)?;
+        serializer.serialize_field(
+            "properties",
+            &SerializedProfilePropertyValue::from_map(self.properties.clone()),
+        )?;
+        serializer.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for GameProfile {
```

Comment:
The auth reply is a list of objects instead of a map and I didn't want to change GameProfile. I guess I could have just impl Serialize/Deserialize and From/Into on an intermediary struct.